app/controllers/api/v1/materials_controller.rb

```diff
@@ -44,6 +44,264 @@ def featured
     render json: materials_data(materials, params[:assigned_to_class])
   end
 
+  #
+  # Get all available materials
+  #
+  def all
+
+    materials = ExternalActivity.includes(:user, :template).all +
+                Interactive.all
+
+    render json: materials_data(materials)
+
+  end
+
+  #
+  # Remove a favorite from the current user.
+  #
+  # Request params should contain:
+  #
+  # favorite_id     The favorite id 
+  #
+  # GET /api/v1/materials/remove_favorite
+  #
+  def remove_favorite
+
+    status  = 200
+    message = "Removing favorite..."
+ 
+    if current_user && !current_user.anonymous?
+      favorite_id = params[:favorite_id]
+
+      if favorite_id
+        favorite = nil
+
+        begin
+          favorite = Favorite.find(favorite_id)
```

Comment:
@scytacki 
There is the "page not found" page, with the concord header and all, that will be displayed if the controller does not catch the exception. But since this is an API call, I think it makes sense to instead return the json with a message rather than an html page indicating an error?